check_pgactivity

```diff
@@ -580,22 +584,27 @@ sub save($$$$) {
 #  * the name of the structure to load
 #  * the path to the file storage
 sub load($$$) {
-    my $host    = shift;
-    my $name    = shift;
-    my $storage = shift;
-    my $hostkey = find_hostkey($host);
+    my $host     = shift;
+    my $name     = shift;
+    my $storage  = shift;
+    my $hostkey  = find_hostkey($host);
+    my $lockfile = "${storage}.lock";
     my $all;
 
     return undef unless -e $storage;
 
-    die "No permission to read status file «${storage}»" unless -r $storage;
+    die "can not read status file «${storage}»" unless -r $storage;
+
+    open my $fh, '<', $lockfile or die "can't open «${lockfile}»: $!";
```

Comment:
Shouldn't the open be `>` instead?  Otherwise it's unnecessary fragile against removal of the lock file.

Reply:
I chose to open in read only because of the following note in the perl documentation of `flock`:

> Note that the fcntl(2) emulation of flock(3) requires that FILEHANDLE be open with read intent to use LOCK_SH and requires that it be open with write intent to use LOCK_EX.

Confirmed from the `man 2 fcntl`:

> In order to place a read lock, fd must be open for reading.  In order to place a write lock, fd must be open for writing.  To place both types of lock, open a file read-write.

So I chose to stay on the compatibility side of the fence, should the implementation of flock rely on fcntl on some user system.

Reply:
Then shouldn't you adapt the rest of the calls accordingly?  For instance on line 574.

Reply:
Sorry I misread the  perldoc part, this is indeed correct :)